cmd/genyaml/gen_kubectl_yaml.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+	"k8s.io/kubernetes/cmd/genutils"
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+type cmdOption struct {
+	Name         string
+	Shorthand    string `yaml:",omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty"`
+	Usage        string `yaml:",omitempty"`
+}
+
+type cmdDoc struct {
+	Name             string
+	Synopsis         string       `yaml:",omitempty"`
+	Description      string       `yaml:",omitempty"`
+	Options          []cmdOption  `yaml:",omitempty"`
+	InheritedOptions []cmdOption	`yaml:"inherited_options,omitempty"`
+	Example          string       `yaml:",omitempty"`
+	SeeAlso          []string     `yaml:"see_also,omitempty"`
+}
+
+func main() {
```

Comment:
Please go for it, that would be awesome!

On Tue, Mar 29, 2016 at 2:23 PM, Thomas Liu <notifications@github.com>
wrote:

> In cmd/genyaml/gen_kubectl_yaml.go
> <https://github.com/kubernetes/kubernetes/pull/23526#discussion_r57803249>
> :
>
> > +	Shorthand    string `yaml:",omitempty"`
> > +	DefaultValue string `yaml:"default_value,omitempty"`
> > +	Usage        string `yaml:",omitempty"`
> > +}
> > +
> > +type cmdDoc struct {
> > +	Name             string
> > +	Synopsis         string       `yaml:",omitempty"`
> > +	Description      string       `yaml:",omitempty"`
> > +	Options          []cmdOption  `yaml:",omitempty"`
> > +	InheritedOptions []cmdOption	`yaml:"inherited_options,omitempty"`
> > +	Example          string       `yaml:",omitempty"`
> > +	SeeAlso          []string     `yaml:"see_also,omitempty"`
> > +}
> > +
> > +func main() {
>
> Sounds good - would making that change to cobra be something I could give
> a try, or were you planning to take it on? Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23526/files/0f08dc4bb7d601f9e2021bebbfa8878af7e201ab..c81f0a7b8222456d74098ffb8cd9cf09531e7a3f#r57803249>
>